cli/src/pcluster/resources/batch/docker/alinux2023/Dockerfile

```diff
@@ -0,0 +1,60 @@
+FROM public.ecr.aws/amazonlinux/amazonlinux:2023
+
+ENV USER root
+
+ARG AWS_REGION
```

Comment:
We're adding alinux2023 as base OS for AWS Batch, did we test it using alinux2023 for batch tests?
If so we should mention this in the changelog and bump the version of awsbatch cli.
Otherwise we can avoid to create this dockerfile.

Reply:
Good point! Thanks Enrico!

Reply:
We tested with Batch and the following integration tests have passed. We will bump the version of Batch CLI during the release process
```
{%- import 'common.jinja2' as common with context -%}
test-suites:
  schedulers:
    test_awsbatch.py::test_awsbatch:
      dimensions:
        - regions: ["us-west-1"]
          instances: {{ common.INSTANCES_DEFAULT_X86 }}
          oss: ["alinux2023"]
          schedulers: ["awsbatch"]
    test_awsbatch.py::test_awsbatch_defaults:
      dimensions:
        - regions: ["ap-northeast-1"]
          instances: {{ common.INSTANCES_DEFAULT_X86 }}
          oss: ["alinux2023"]
          schedulers: ["awsbatch"]
```